### PR TITLE
feat: add raw_thread_delete, raw_thread_update, and raw_thread_member_remove events

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

add raw_thread_delete, raw_thread_update, and raw_thread_member_remove events

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by bast0006 — APPROVED** [review]

File: docs/api.rst [hunk 1]
```diff
@@ -892,23 +894,57 @@ to handle it, which defaults to print a traceback and ignoring the exception.
     :param thread: The thread that got deleted.
     :type thread: :class:`Thread`
 
+.. function:: on_raw_thread_delete(payload)
+
+    Called whenever a thread is deleted.
+    Unlike :func:`on_thread_update`, this is called
```
Comment (shiftinv) [inline docs/api.rst:900]:
```suggestion
    Unlike :func:`on_thread_delete`, this is called
```

File: docs/api.rst [hunk 2]
```diff
@@ -919,6 +955,19 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
+
+    Called whenever a thread is updated.
+    Unlike :func:`on_thread_update`, this is called
+    regardless of the state of the internal thread cache.
+
+    This requires :attr:`Intents.guilds` to be enabled.
+
+    .. versionadded:: 2.5
+
+    :param thread: The updated thread's new info.
```
Comment (shiftinv) [inline docs/api.rst:968]:
```suggestion
    :param thread: The updated thread.
```

File: disnake/raw_models.py [hunk 3]
```diff
@@ -306,6 +310,68 @@ def __init__(self, data: GuildScheduledEventUserActionEvent):
...
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+    thread_id: :class:`int`
+        The ID of the thread that was deleted.
+    guild_id: :class:`int`
+        The ID of the guild the thread was deleted in.
+    thread_type: :class:`ChannelType`
+        The channel type of the deleted thread.
```
Comment (shiftinv) [inline disnake/raw_models.py:325]:
```suggestion
        The type of the deleted thread.
```

File: docs/api.rst (hunk 1 above)
Comment (onerandomusername) [reply docs/api.rst]: https://github.com/DisnakeDev/disnake/pull/495/commits/d23d7cf63f41d0396d1572bf879fbab3862365a7

File: docs/api.rst (hunk 2 above)
Comment (onerandomusername) [reply docs/api.rst]: https://github.com/DisnakeDev/disnake/pull/495/commits/d23d7cf63f41d0396d1572bf879fbab3862365a7

File: disnake/raw_models.py (hunk 3 above)
Comment (onerandomusername) [reply disnake/raw_models.py]: https://github.com/DisnakeDev/disnake/pull/495/commits/d23d7cf63f41d0396d1572bf879fbab3862365a7

**Review by shiftinv — APPROVED** [review]
lgtm

**Pull request merged**